### PR TITLE
Unified SupplyPolicy type

### PR DESCRIPTION
**Pull request by nmboavida**
Removed Collection Cap generic
Collection now has SupplyPolicy object

### CODE REVIEW

File: sources/collection/std_collection.move [hunk 1]
```diff
@@ -46,7 +45,7 @@ module nft_protocol::std_collection {
         // Symbol of the Nft Collection. This parameter is a
         // vector of bytes that should enconde to utf8
         symbol: vector<u8>,
-        max_supply: Option<u64>,
+        max_supply: u64,
```
Comment (bausano) [inline sources/collection/std_collection.move:48]: Unlimited is done with `u64::MAX`?

Comment (bausano) [reply sources/collection/std_collection.move:48]: Nope, that's not the case upon reading the next changed LoC

File: sources/supply/policy.move [hunk 2]
```diff
@@ -0,0 +1,70 @@
+//! Module contaning two supply `Cap` types, namely `Limited` and `Unlimited`.
```
Comment (bausano) [inline sources/supply/policy.move:1]: Perhaps change this doc to refer to the `is_blind` flag.

Re: flag - not sure whether `is_blind` is a good name. Did you have some other ideas which you discarded in favour of `is_blind`?

Comment (nmboavida) [reply sources/supply/policy.move:1]: I renamed it to `regulated` which inverts the logic. I made the necessary changes in all the conditions.

File: sources/collection/std_collection.move (hunk 1 above)
Comment (nmboavida) [reply sources/collection/std_collection.move:48]: To create a collection with unregulated supply the caller will set `max_supply = 0`. If `max_supply > 0` then the caller will be creating a collection with regulated supply. I added some documentation comments explaining the logic.

**Pull request merged**